src/main/java/org/jvnet/hudson/test/HudsonHomeLoader.java

```diff
@@ -121,11 +126,31 @@ public File allocate() throws Exception {
             return new CopyExisting(home).allocate();
         }
 
+        public static boolean isJavaIdentifier(@CheckForNull String name) {
+            if (StringUtils.isEmpty(name)) {
+                return false;
+            }
+            if (!Character.isJavaIdentifierStart(name.charAt(0))) {
+                return false;
+            }
+            for (int pos = 1; pos < name.length(); ++pos) {
+                if (!Character.isJavaIdentifierPart(name.charAt(pos))) {
+                    return false;
+                }
+            }
+            return true;
+        }
+
         private URL findDataResource() {
             // first, check method specific resource
             Class<?> clazz = testMethod.getDeclaringClass();
+            String methodName = testMethod.getName();
+            
+            if (isJavaIdentifier(alterName)) {
```

Comment:
Why do we care whether it is a Java identifier or not? Should not matter for purposes of resource loading. Anyway, if it is unusable for some reason, we should fail with an error.